drivers/flash/Kconfig.stm32_qspi

```diff
@@ -0,0 +1,21 @@
+# STM32 Quad SPI flash driver configuration options
+
+# Copyright (c) 2020 Piotr Mienkowski
+# Copyright (c) 2020 Linaro Limited
+# SPDX-License-Identifier: Apache-2.0
+
+DT_COMPAT_ST_STM32_QSPI_NOR := st,stm32-qspi-nor
+
+config FLASH_STM32_QSPI
+	bool "STM32 Quad SPI Flash driver"
+	depends on SOC_FAMILY_STM32
+	default $(dt_compat_enabled,$(DT_COMPAT_ST_STM32_QSPI_NOR))
+	select USE_STM32_HAL_QSPI
+	select USE_STM32_HAL_DMA
```

Comment:
Should it be automatic with optional DMAs? Currently DMA for H7 does not even compile...

Reply:
Txs for finding this. It is now made conditional on dts status.